akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala

```diff
@@ -167,6 +167,8 @@ private[http] class HttpRequestParser(_settings: ParserSettings,
             emitRequestStart(emptyEntity(cth))
             setCompletionHandling(HttpMessageParser.CompletionOk)
             startNewMessage(input, bodyStart)
+          } else if (!method.isEntityAccepted) {
+            failMessageStart(UnprocessableEntity, s"${method.name} requests must not have an entity")
```

Comment:
Should the `isEntityAccepted` check get removed [here](https://github.com/akka/akka/blob/release-2.3-dev/akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpMessage.scala#L144) then?